eZ/Publish/Core/Repository/TrashService.php

```diff
@@ -107,13 +108,18 @@ public function loadTrashItem($trashItemId)
      * Content is left untouched.
      *
      * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException if the user is not allowed to trash the given location
+     * @throws \eZ\Publish\API\Repository\Exceptions\ForbiddenException if the user is trying trash the root location
      *
      * @param \eZ\Publish\API\Repository\Values\Content\Location $location
      *
      * @return null|\eZ\Publish\API\Repository\Values\Content\TrashItem null if location was deleted, otherwise TrashItem
      */
     public function trash(Location $location)
     {
+        if ($location->depth === 1) {
```

Comment:
well, it's just to know if i should still go for depth check or not, and if not, if we change what's donde on platform ui for consistency :)